### PR TITLE
[Draft] Reusing the context in chaosengine controller

### PR DESCRIPTION
**Pull request by rajdas98**
Signed-off-by: Raj Das <mail.rajdas@gmail.com>
- Currently, empty ctx is being created and not being reusing in the chaos engine.

**What this PR does / why we need it**:

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:

**Checklist:**
- [ ] Fixes #<issue number>
- [ ] Labelled this PR & related issue with `documentation` tag
- [ ] PR messages has document related information
- [ ] Labelled this PR & related issue with `breaking-changes` tag
- [ ] PR messages has breaking changes related information
- [ ] Labelled this PR & related issue with `requires-upgrade` tag
- [ ] PR messages has upgrade related information
- [ ] Commit has unit tests
- [ ] Commit has integration tests

### CODE REVIEW

**Comment by rahulchheda** [comment]
@rajdas98 Can you point to the issue this PR is solving?

**Comment by rajdas98** [comment]
> @rajdas98 Can you point to the issue this PR is solving?

There is no issue for this. Currently we are not using context propoperly. We just instantiate `TODO` (dummy) context everytime when we call it.

You can check this for more details- https://medium.com/@cep21/how-to-correctly-use-context-context-in-go-1-7-8f2c0fafdf39#:~:text=Context%20should%20flow%20through%20your%20program&text=Nor%20do%20you%20want%20to,when%20the%20request%20is%20over.

**Comment by rahulchheda** [comment]
> > @rajdas98 Can you point to the issue this PR is solving?
> 
> There is no issue for this. Currently we are not using context properly. We just instantiate `TODO` (dummy) context every time when we call it.
> 
> You can check this for more details- https://medium.com/@cep21/how-to-correctly-use-context-context-in-go-1-7-8f2c0fafdf39#:~:text=Context%20should%20flow%20through%20your%20program&text=Nor%20do%20you%20want%20to,when%20the%20request%20is%20over.

Yes, we are using context.TODO() everywhere, that is because the operation that we are performing in the controller doesn't support request cancellation. Context mainly is used for cancellation propagation, but in our case, that is not needed.

File: pkg/controller/chaosengine/chaosengine_controller.go [hunk 1]
```diff
@@ -62,6 +62,7 @@ type ReconcileChaosEngine struct {
 	// recorder is an event recorder for recording Event resources to the
 	// Kubernetes API.
 	recorder record.EventRecorder
+	ctx context.Context
```
Comment (akhilerm) [inline pkg/controller/chaosengine/chaosengine_controller.go:65]: Where is this `ctx` variable set.? While creating the struct you can set it as may be `context.TODO()`  or `context.Background()`

**Pull request closed without merging**